frontend/src/components/Organization.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import * as api from '../api';
+import { User } from '../services/auth';
+import Errors from './Errors';
+import Loader from './Loader';
+
+interface OrgProps {
+  user: User;
+  orgName: string;
+}
+
+interface OrgState {
+  done: boolean;
+  org: api.OrgResponse | undefined;
+  errors: string[];
+
+  config: string;
+}
+
+class Organization extends React.Component<OrgProps, OrgState> {
+  constructor(props: OrgProps) {
+    super(props);
+
+    this.state = {
+      done: false,
+      org: undefined,
+      errors: [],
+      config: ''
+    };
+
+    this.handleConfigChange = this.handleConfigChange.bind(this);
+    this.handleConfigSave = this.handleConfigSave.bind(this);
+  }
+
+  public componentDidMount() {
+    return api
+      .org(this.props.orgName)
+      .then(resp =>
+        this.setState({
+          done: true,
+          org: resp,
+          errors: [],
+          config: resp.config
+        })
+      )
+      .catch(err => {
+        this.setState({
+          done: true,
+          org: undefined,
+          errors: err,
+          config: ''
+        });
+      });
+  }
+
+  public render() {
+    if (!this.state.done) {
+      return <Loader />;
+    }
+
+    const org = this.state.org;
+
+    if (this.state.errors.length > 0 || org === undefined) {
+      return <Errors errors={this.state.errors} />;
+    }
+
+    return (
+      <div>
+        <h1>Settings for Organization {org.name}</h1>
+        <textarea
+          value={this.state.config}
+          onChange={this.handleConfigChange}
+          style={{ font: 'monospace', width: '300px', height: '10em' }}
+        />
+        <div>
+          <br />
+          <button onClick={this.handleConfigSave}>Save</button>
+        </div>
+      </div>
+    );
+  }
+
+  private handleConfigChange(event: React.ChangeEvent<HTMLTextAreaElement>) {
+    this.setState({ config: event.target.value });
+  }
+
+  private handleConfigSave() {
+    if (this.state.org === undefined) {
+      // tslint:disable-next-line
+      console.error('handleConfigSave called with undefined state.org');
```

Comment:
actually, it looks like we should throw an exception here. And later we can catch it using [error boundaries](https://reactjs.org/docs/error-boundaries.html) if we want instead of a silent broken button.

Reply:
It seems [error boundaries do not apply in handlers](https://reactjs.org/docs/error-boundaries.html#how-about-event-handlers), but I followed the example to render the error in 341048e.